pkg/reconciler/common/install_test.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	mf "github.com/manifestival/manifestival"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"knative.dev/operator/pkg/apis/operator/v1alpha1"
+)
+
+func TestInstall(t *testing.T) {
+	// Resources in the manifest
+	version := "v0.14-test"
+	deployment := NamespacedResource("apps/v1", "Deployment", "test", "test-deployment")
+	role := NamespacedResource("rbac.authorization.k8s.io/v1", "Role", "test", "test-role")
+	roleBinding := NamespacedResource("rbac.authorization.k8s.io/v1", "RoleBinding", "test", "test-role-binding")
+	clusterRole := ClusterScopedResource("rbac.authorization.k8s.io/v1", "ClusterRole", "test-cluster-role")
+	clusterRoleBinding := ClusterScopedResource("rbac.authorization.k8s.io/v1", "ClusterRoleBinding", "test-cluster-role-binding")
+
+	// Deliberately mixing the order in the manifest.
+	in := []unstructured.Unstructured{*deployment, *role, *roleBinding, *clusterRole, *clusterRoleBinding}
+	// Expect things to be applied in order.
+	want := []*unstructured.Unstructured{role, clusterRole, roleBinding, clusterRoleBinding, deployment}
+
+	client := &fakeClient{}
```

Comment:
I should publish a `mf.fake.Client`, I think. I've wanted one more than once.

Reply:
Not quite sure if that's necessary. A little helper that makes me only implement the functions really under test would be nice maybe.